packages/transactions/src/TransactionManager.ts

```diff
@@ -483,7 +483,7 @@ class TransactionManager {
     const connectedWallet = smartAccountState.address
     walletContract = walletContract.attach(connectedWallet)
 
-    const isDeployed = smartAccountState.isDeployed
+    const isDeployed = await this.contractUtils.isDeployed(chainId, smartAccountState.address)
```

Comment:
how is this reflected / needed?

Reply:
Previously, the batching function decided whether to fetch the nonce from the proxy or just use 0 based on the value of smartAccountState.isDeployed. If a new transaction was sent before resetting the SDK, right after deploying the wallet, this could lead to an invalid nonce error.